src/ray/raylet/scheduling/fair_scheduling_queue_test.cc

```diff
@@ -0,0 +1,142 @@
+#include "ray/raylet/scheduling/fair_scheduling_queue.h"
+#include "ray/raylet/scheduling/test_util.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace ray {
+
+namespace raylet {
+
+using ::testing::_;
+
+class FairSchedulingQueueTest : public ::testing::Test {
+ public:
+  FairSchedulingQueue queue_;
+
+  Work TaskToWork(Task &task) { return std::make_tuple(task, nullptr, nullptr); }
+
+  void AssertNoLeaks() {
+    ASSERT_EQ(queue_.active_tasks_.size(), 0);
+    ASSERT_EQ(queue_.work_queue_.size(), 0);
+  }
+};
+
+TEST_F(FairSchedulingQueueTest, TestBasic) {
+  /*
+   * When tasks are queued, make sure we prioritize the shapes that aren't
+   * currently running.
+   */
+  Task task = CreateTask({{ray::kCPU_ResourceLabel, 4}});
+  Task task1 = CreateTask({{ray::kCPU_ResourceLabel, 2}});
+  Task task2 = CreateTask({{ray::kCPU_ResourceLabel, 1}});
+
+  queue_.MarkRunning(task);
+  queue_.MarkRunning(task);
```

Comment:
typo? marking running twice for the same key?

Reply:
I think he wanted to simulate two of 4 cpus shapes are running concurrently?